src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/FindAllReferencesHandler.cs

```diff
@@ -170,6 +170,11 @@ private async Task<VSReferenceItem[]> RemapReferenceItemsAsync(VSReferenceItem[]
                     continue;
                 }
 
+                // Temporary fix for codebehind leaking through
+                // Revert when https://github.com/dotnet/aspnetcore/issues/22512 is resolved
+                referenceItem.DefinitionText = FilterReferenceDisplayText(referenceItem.DefinitionText);
```

Comment:
So basically the idea here is that we want to do this filtering earlier so that it catches the `!RazorLSPConventions.IsRazorCSharpFile` case?

Reply:
Exactly. We want to ensure we do filtering for all results, including `.razor` files, as roslyn now returns mapped results instead of locations in generated files.